### PR TITLE
22224-ZipArchiveMember-still-uses-deprecated-RWBinaryOrTextStream-and-MultiByteBinaryOrTextStream

### PR DESCRIPTION
**Pull request by VincentBlondeau**
Add tests + convert the deprecated streams to MemoryFileSystemFile

https://pharo.fogbugz.com/f/cases/22224/ZipArchiveMember-still-uses-deprecated-RWBinaryOrTextStream-and-MultiByteBinaryOrTextStream

### CODE REVIEW

File: src/Compression/ZipArchiveMember.class.st [hunk 1]
```diff
@@ -205,33 +205,30 @@ ZipArchiveMember >> contentStreamFromEncoding: encodingName [
...
 
-	| s |
-	s := MultiByteBinaryOrTextStream on: (String new: self uncompressedSize).
-	s converter:  (TextConverter newForEncoding: encodingName).
-	self extractTo: s.
-	s reset.
-	^ s
-
+	| file |
+	file := MemoryFileSystemFile named: 'tempExtractFile'.
```
Comment (pavel-krivanek) [inline src/Compression/ZipArchiveMember.class.st:209]: The FileSystem-Memory package is not part of the basic system. It is loaded with Metacello but Zip support is already needed for loading packages using Monticello. That's why the build fails

Comment (svenvc) [reply src/Compression/ZipArchiveMember.class.st:209]: I have not tried, but would

```suggestion
    ^ ByteArray writeStreamDo: [ :stream | self extractTo: stream ] 
```

not work ?

Comment (guillep) [reply src/Compression/ZipArchiveMember.class.st:209]: I think so too

**Comment by guillep** [comment]
I don't think making Zip use MemoryFileSystem is ok. We should just use normal streams on byte arrays.

**Comment by guillep** [comment]
I'm working on it

**Comment by guillep** [comment]
I've redone the PR using regular streams on bytearray and strings.

https://github.com/pharo-project/pharo/pull/2180

**Pull request closed without merging**